tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ManagedIdentityTests.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Castle.Core.Internal;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Json;
+using Microsoft.Identity.Test.Integration.NetFx.Infrastructure;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NSubstitute.Exceptions;
+
+namespace Microsoft.Identity.Test.Integration.HeadlessTests
+{
+    [TestClass]
+    public class ManagedIdentityTests
+    {
+        private static readonly string[] s_msi_scopes = { "https://management.azure.com" };
+        private static readonly string s_baseURL = "https://service.msidlab.com/";
+        private static readonly string s_clientId = "client_id";
+        private static Dictionary<string, string> s_envVariables = new Dictionary<string, string>();
+        private const string UserAssignedClientID = "3b57c42c-3201-4295-ae27-d6baec5b7027";
+        private const string Mi_res_id = "/subscriptions/c1686c51-b717-4fe0-9af3-24a20a41fb0c/resourcegroups/MSAL_MSI/providers/Microsoft.ManagedIdentity/userAssignedIdentities/MSAL_MSI_USERID";
+
+        public enum AzureResource
+        {
+            webapp,
+            function,
+            vm,
+            azurearc,
+            cloudshell,
+            servicefabric
+        }
+
+        [TestMethod]
+        public async Task ManagedIdentitySourceCheckAsync()
+        {
+            //Arrange
+            string result = string.Empty;
+            string exception = string.Empty;
+            string expectedClientException = "Authentication with managed identity is unavailable. " +
+                "No managed identity endpoint found.";
+
+            IConfidentialClientApplication cca = CreateCCAWithProxy(s_baseURL);
+
+            //Act
+            try
+            {
+                AuthenticationResult authenticationResult = await cca.AcquireTokenForClient(s_msi_scopes)
+                    .WithManagedIdentity()
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+            }
+            catch (MsalClientException ex)
+            {
+                result = ex.Message;
+            }
+            catch (Exception ex)
```

Comment:
We should not catch general exceptions here only MsalClientException

Reply:
resolved